### PR TITLE
Flutter for Xamarin.Forms

### PR DESCRIPTION
**Pull request by adamped**
Document based off Flutter for Android and Flutter for iOS, designed to help Xamarin.Forms developers get up to speed with Flutter.

This is the first draft.

Please let me know if there are any changes or enhancements needed.

### CODE REVIEW

**Comment by adamped** [comment]
I signed it!

**Comment by Sfshaza** [comment]
@adamped, I'm starting to review this. Can you provide a staged version?

File: flutter-for-xamarin-forms.md [hunk 1]
```diff
@@ -0,0 +1,2322 @@
...
+
+In Android, you add dependencies by adding to your Gradle build script. In iOS, you add 
+dependencies by adding to your `Podfile`.
+
+Flutter uses Dart's own build system, and the Pub package manager.
+The tools delegate the building of the native Android and iOS wrapper apps to the
+respective build systems.
+
+In general, use `pubspec.yaml` to declare external dependencies to use in Flutter. A good 
+place to find Flutter packages is [Pub](https://pub.dartlang.org/flutter/packages/).
```
Comment (Sfshaza) [inline flutter-for-xamarin-forms.md]: Ha! Another example where I added this above.

I would use pub.dartlang.org/flutter as the link.

File: flutter-for-xamarin-forms.md [hunk 2]
```diff
@@ -0,0 +1,2322 @@
...
+
+## How do I listen to application lifecycle events?
+
+In Xamarin.Forms, you have an `Application` that contains `OnStart`, `OnResume` and
+`OnSleep`. In Flutter you can instead listen to similar lifecycle events by hooking into 
+the `WidgetsBinding` observer and listening to the `didChangeAppLifecycleState()` change event.
+
+The observable lifecycle events are:
+
+* `inactive` — The application is in an inactive state and is not receiving
```
Comment (Sfshaza) [inline flutter-for-xamarin-forms.md]: It's enough to say "This event is iOS only".

File: flutter-for-xamarin-forms.md [hunk 3]
```diff
@@ -0,0 +1,2322 @@
...
+user input. This event only works on iOS, as there is no equivalent event on
+Android.
+* `paused` — The application is not currently visible to
+the user, is not responding to user input, but is running in the background.
+* `resumed` — The application is visible and responding to user input.
+* `suspending` — The application is suspended momentarily. The iOS platform
```
Comment (Sfshaza) [inline flutter-for-xamarin-forms.md]: "This event is Android only."

File: flutter-for-xamarin-forms.md [hunk 4]
```diff
@@ -0,0 +1,2322 @@
...
+        left: 10.0,
+        child: new Icon(Icons.add_circle, size: 24.0, color: const Color.fromRGBO(0,0,0,1.0)),
+      ),
+    ],
+  ),
+{% endprettify %}
+
+## What is the equivalent of a ScrollView?
+
+In Xamarin.Forms you use a ScrollView wrap around a `VisualElement` and if the content is
```
Comment (Sfshaza) [inline flutter-for-xamarin-forms.md]: In Xamarin.Forms, a ScrollView wraps around a `VisualElement` and, if the content is larger than the device screen, it scrolls.

File: flutter-for-xamarin-forms.md [hunk 5]
```diff
@@ -0,0 +1,2322 @@
...
+        new Text('Row Three'),
+        new Text('Row Four'),
+      ],
+    );
+  }
+{% endprettify %}
+
+## How do I handle landscape transitions in Flutter?
+
+FlutterView handles the config change if AndroidManifest.xml contains:
```
Comment (Sfshaza) [inline flutter-for-xamarin-forms.md]: What's a "FlutterView"?

File: flutter-for-xamarin-forms.md [hunk 6]
```diff
@@ -0,0 +1,2322 @@
...
+Instead of creating a "ListView", create a ListView.builder that
+takes two key parameters: the initial length of the list, and an ItemBuilder
+function.
+
+The ItemBuilder function is similar to the `getView` function in an Android
+adapter; it takes a position, and returns the row you want rendered at
+that position.
+
+Finally, but most importantly, notice that the `onTap()` function
+doesn't recreate the list anymore, but instead `.add`s to it.
```
Comment (Sfshaza) [inline flutter-for-xamarin-forms.md]: Please link to the codelab on this topic: (Though actually only part 1 builds the infinite list.)

[Write your first Flutter App, part 1](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt1/index.html?index=..%2F..%2Findex#0)
and
[Write Your First Flutter App, part 2](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt2/index.html?index=..%2F..%2Findex#0)

**Comment by adamped** [comment]
thanks @Sfshaza 

Though for the newbie here, what is involved in providing a staged version?

**Comment by Sfshaza** [comment]
Oh sorry! We used to have this as part of the README.

Build the website:  bundle exec jekyll build
Then deploy it to a project on Firebase: firebase deploy
  (Once you've specified which FB project, such as "firebase use
<myproject>.

You should be able to set up a free FB account and create a project through
their website.

**Comment by Sfshaza** [comment]
@Hixie, can you explain this? The _includes/sidebar.html file diff shows the entirety of both files, rather than the one line added. I saved each off as a raw file and ran diff and the only difference is that one file. Both files have the same name.  Can you explain it?

File: flutter-for-xamarin-forms.md [hunk 7]
```diff
@@ -0,0 +1,2322 @@
...
+Navigate to a route by `push`ing its name to the `Navigator`.
+
+<!-- skip -->
+{% prettify dart %}
+Navigator.of(context).pushNamed('/b');
+{% endprettify %}
+
+The `Navigator` class handles routing in Flutter and is used to get
+a result back from a route that you have pushed on the stack. This is done
+by `await`ing on the `Future` returned by `push()`.
```
[the remark this answers is not in the document]
Comment (adamped) [reply flutter-for-xamarin-forms.md]: I'll add a link in to show how the syntax slightly changes. Async/await is very common in .NET and Dart's version is almost identical. It is a simple thing for .NET dev's to grasp :)

**Comment by adamped** [comment]
@Sfshaza - thank you for your feedback. I have corrected everything you mention. Apologies on the delay, and apologies on my use of passive voice. I continually trip up on it.

I have created a staging environment: https://flutterwebsite.firebaseapp.com/flutter-for-xamarin-forms/.

**Comment by Sfshaza** [comment]
Thanks so much, @adamped! You are a good sport. I'm going to merge this PR, and will clean up any minor copyediting issues later. (There's always somethin'.)  Thanks again!

**Pull request merged**